modules/solace/src/test/java/org/testcontainers/containers/SolaceContainerSMFTest.java

```diff
@@ -0,0 +1,167 @@
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author Tomasz Forys
+ */
+public class SolaceContainerSMFTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SolaceContainerSMFTest.class);
+
+    private static final String MESSAGE = "HelloWorld";
+
+    private static final Topic TOPIC = JCSMPFactory.onlyInstance().createTopic("Topic/ActualTopic");
+
+    @Test
+    public void testSolaceContainerWithSimpleAuthentication() {
+        try (
+            // solaceContainerSetup {
+            SolaceContainer solaceContainer = new SolaceContainer("solace/solace-pubsub-standard:10.2")
+                .withCredentials("user", "pass")
+                .withTopic("Topic/ActualTopic", Service.SMF)
+                .withVpn("test_vpn")
+            // }
+        ) {
+            solaceContainer.start();
+            JCSMPSession session = createSessionWithBasicAuth(solaceContainer);
+            Assertions.assertThat(session).isNotNull();
+            Assertions.assertThat(consumeMessageFromSolace(session)).isEqualTo(MESSAGE);
+            session.closeSession();
+        }
+    }
+
+    @Test
+    public void testSolaceContainerWithCertificates() {
+        try (
+            // solaceContainerUsageSSL {
+            SolaceContainer solaceContainer = new SolaceContainer("solace/solace-pubsub-standard:10.2")
+                .withClientCert(
+                    MountableFile.forClasspathResource("solace.pem"),
+                    MountableFile.forClasspathResource("rootCA.crt")
+                )
+                .withTopic("Topic/ActualTopic", Service.SMF_SSL)
+            // }
+        ) {
+            solaceContainer.start();
+            JCSMPSession session = createSessionWithCertificates(solaceContainer);
+            Assertions.assertThat(session).isNotNull();
+            Assertions.assertThat(consumeMessageFromSolace(session)).isEqualTo(MESSAGE);
+            session.closeSession();
+        }
+    }
+
+    private String getResourceFileLocation(String name) {
+        return Thread.currentThread().getContextClassLoader().getResource(name).getPath();
+    }
+
+    private static JCSMPSession createSessionWithBasicAuth(SolaceContainer solace) {
+        JCSMPProperties properties = new JCSMPProperties();
+        properties.setProperty(JCSMPProperties.HOST, solace.getOrigin(Service.SMF));
+        properties.setProperty(JCSMPProperties.VPN_NAME, solace.getVpn());
+        properties.setProperty(JCSMPProperties.USERNAME, solace.getUsername());
+        properties.setProperty(JCSMPProperties.PASSWORD, solace.getPassword());
+        return createSession(properties);
+    }
+
+    private JCSMPSession createSessionWithCertificates(SolaceContainer solace) {
+        JCSMPProperties properties = new JCSMPProperties();
+        properties.setProperty(JCSMPProperties.HOST, solace.getOrigin(Service.SMF_SSL));
+        properties.setProperty(JCSMPProperties.VPN_NAME, solace.getVpn());
```

Comment:
Can we add the following property? Otherwise, it will fail with `Error connecting and setting up session! CertificateException - No subject alternative names matching IP address 127.0.0.1 found` in my case. We can add a note `Just for testing purposes`

```suggestion
        properties.setProperty(JCSMPProperties.SSL_VALIDATE_CERTIFICATE_HOST, false);
        properties.setProperty(JCSMPProperties.VPN_NAME, solace.getVpn());
```